### PR TITLE
feat: Make CHECKSUM and TIMESTAMP vars available in cmds commands

### PR DESCRIPTION
**Pull request by niklasr22**
Hello, this is my attempt to implement #1735.

### CODE REVIEW

File: variables.go [hunk 1]
```diff
@@ -228,7 +181,7 @@ func (e *Executor) compiledTask(call *ast.Call, evaluateShVars bool) (*ast.Task,
 		}
 	}
 
-	if len(origTask.Status) > 0 {
```
Comment (vmaerten) [inline variables.go:231]: I'm not a big fan of those two nested if statements.
`TIMESTAMP` and `CHECKSUM` should only be available if there are sources defined. I'd replace `len(origTask.Status) > 0` with `len(origTask.Source) > 0` and move it above `cmds`.

**Review by vmaerten — COMMENTED** [review]
Hello
Thanks for your PR, I've left some comments!

File: variables.go (hunk 1 above)
Comment (niklasr22) [reply variables.go:231]: True, good idea. I changed it like you said but now it also has the effect that .TIMESTAMP and .CHECKSUM are available in preconditions and deps. Do you want to keep it like that?

Comment (vmaerten) [reply variables.go:231]: Indeed this is wanted :slightly_smiling_face:

**Comment by vmaerten** [comment]
Lint job is failing. a `t.Parallele` is missing in your test
After that it should be good

**Review by vmaerten — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**